TurnkeyReact/Oauth.tsx

```diff
@@ -0,0 +1,143 @@
+import { useTurnkey } from "@turnkey/sdk-react-native";
+import { useEffect, useState, useCallback } from "react";
+import { SharedConfigs } from "./SharedConfigs";
+import { Button } from "./ui/button";
+import { View, Text } from "react-native";
+// import GoogleIcon from "../../assets/svgs/google.svg";
+import { OAUTH_TOKEN_EXPIRATION_SECONDS } from "./lib/constants";
+// import * as Crypto from "expo-crypto";
+import { styles } from "./TurnkeyStyle";
+import SHA256 from 'crypto-js/sha256';
+
+interface OAuthProps {
+  onSuccess: (params: {
+    oidcToken: string;
+    providerName: string;
+    targetPublicKey: string;
+    expirationSeconds: string;
+  }) => Promise<void>;
+  configs: SharedConfigs
+}
+
+interface AuthButtonProps extends OAuthProps {
+  nonce: string | null;
+  targetPublicKey: string | null;
+  refreshNonce: () => Promise<void>;
+}
+
+export const GoogleAuthButton: React.FC<AuthButtonProps> = ({
+  onSuccess,
+  nonce,
+  targetPublicKey,
+  refreshNonce,
+  configs,
+}) => {
+  const { handleGoogleOAuth } = useTurnkey();
+
+  const onIdToken = async (idToken: string) => {
+    await onSuccess({
+      oidcToken: idToken,
+      providerName: "google",
+      targetPublicKey: targetPublicKey!,
+      expirationSeconds: OAUTH_TOKEN_EXPIRATION_SECONDS,
+    });
+
+    // we refresh the nonce before authentication to ensure a new one is used
+    // if the user logs out and logs in with oAuth again
+    await refreshNonce();
+  };
+
+  const handlePress = async () => {
+    try {
+      await handleGoogleOAuth({
+        clientId: configs.googleClientId,
+        nonce: nonce!,
+        scheme: configs.appScheme,
+        onSuccess: onIdToken,
+      });
+    } catch (error) {
+      console.error("Error in Google Auth:", error);
+    }
+  };
+
+  return (
+    <Button
+      onPress={handlePress}
+     // className="border border-black rounded-xl bg-transparent flex-row items-center justify-center flex-1 h-16"
+      disabled={!nonce || !targetPublicKey}
```

Comment:
This will be disabled if `nonce` is zero. double checking if intended behavior. otherwise can use `nonce == null`